state/stateStore.js

```
@@ -1,48 +1,40 @@
-import { refreshToken } from './token.js';
-
 export class StateStore {
 	constructor() {
 		this._states = new Map();
 	}
 
-	async add(state) {
-		await state.refreshToken();
-		if (!state || !state.entityId || !state.token.toString()) {
+	add(state) {
+		if (!state || !state.entityID || !state.token.toString()) {
 			return;
 		}
 
-		this._initContainer(state.entityId, state.token);
+		const tokenMap = this._initTokenMap(state.token);
+		tokenMap.set(state.entityID, state);
 	}
 
-	async get(entityID, token) {
-		await refreshToken(token);
-		const lowerCaseEntityId = entityID.toLowerCase();
+	get(entityID, token) {
+		const lowerCaseEntityID = entityID.toLowerCase();
 		const tokenCache = token.toString();
 
-		const state = this.has(entityID, token) && this._states.get(tokenCache).get(lowerCaseEntityId).keys().next().value;
-		state.refreshToken();
+		const state = this.has(entityID, token) && this._states.get(tokenCache).get(lowerCaseEntityID);
+
 		return state;
 	}
 
-	async has(entityID, token) {
-		await refreshToken(token);
-		const lowerCaseEntityId = entityID.toLowerCase();
+	has(entityID, token) {
+		const lowerCaseEntityID = entityID.toLowerCase();
```

can entity or token ever be null or undefined? get and has methods are just assuming good data atm

---

Yeah, if we this gets bad data then it will send an error to the browser saying `entityID` is null and `toLowerCase` can't be called.

If I caught this I would throw the same error. So assuming good data here is (in my opinion) what we should do. Adding checks makes the code more messy and there is no security risk with bad data here.

However, this is strictly in my opinion and convention would say I am wrong. So it probably pretty safe to assume I am wrong. Can someone explain the value it would add by checking?

---

Yeah, in both those cases you would get something like: `cannot get property "toLowerCase" of undefined/null`, which I think is descriptive enough of an error for devs. Trying to get something out of a `Map` with a null value is not something I would expect, so that default uncaught error is exactly what it should throw IMO

---

It may not be necessary then, I am looking at the file as an individual class. If any potential errors are being handled elsewhere and this is only used internally, then we understand the implications. Then just need to make sure we have tests only those other systems to make sure it doesn't break.

---

So, we did send an empty `entityID` when I first ran this code from a bug. It was good to check.